docs/development/new_component_onboarding.md

```diff
@@ -0,0 +1,71 @@
+# Onboarding a New Component into TheRock  
+*(For new component teams — use this as your checklist and reference)*
+
+## Purpose  
+This document describes the steps for bringing a new component into TheRock (ROCm build platform) from initial commit through CI integration, using the example of a previous component add-up.
+
+## Pre-Requisites  
+Before you start, ensure you have:  
+- A publicly accessible upstream repository for your component (cloneable).  
+- A reproducible build (preferably CMake) of your component.  
+- A test suite (smoke and full/regression) that can be run non-interactively in CI.  
+- Any patches needed for TheRock integration (packaged as `.patch` files).  
+- Defined artifact outputs (what binaries or libraries will be produced).  
+- A point of contact in your team for component maintenance.
+
+## Step-by-Step Process  
+
+### 1. Add the source into TheRock  
+- Add your component under `math-libs/<component-name>` (or appropriate category) as a submodule or source directory.  
+- Update `.gitmodules` accordingly if using submodules.  
+- Commit with message: `Add <component-name> submodule`.
+
+### 2. Create artifact descriptor  
+- Create `artifact-<component-name>.toml` in the same category directory.  
+- Define the artifacts your build will generate and how TheRock should fetch or publish them.
+
+### 3. Integrate into CMake build graph  
+- Update `CMakeLists.txt` in the category folder (e.g., `math-libs/CMakeLists.txt`) to add your component with `add_subdirectory()`.  
+- If needed, update the top-level `CMakeLists.txt` to include your category changes.  
+- Provide an `examples/CMakeLists.txt` if your component has sample builds.
+
+### 4. Update fetch/checkout logic  
+- Edit `build_tools/fetch_sources.py` (and/or related scripts) to include your component so TheRock can fetch the correct commit/tag upstream.  
+- Specify correct version/branch and any special fetch instructions.
+
+### 5. Add CI test script  
+- Under `build_tools/github_actions/test_executable_scripts/`, add `test_<component-name>.py`.  
+- The script should support:  
+  - A “smoke” mode (quick > 5-10 mins)  
+  - A “full” mode (longer, sharded)  
+  - Timeout parameters from environment variables or CI config.  
+- Ensure invocation is non-interactive and returns proper exit codes on failure.
```

Comment:
This overlaps with https://github.com/ROCm/TheRock/blob/main/docs/development/adding_tests.md.

Same comments as for others - avoid duplicating documentation (this may involve moving the existing docs around too, and that's fine).